gui/mod-manager.lua

```diff
@@ -401,6 +402,120 @@ function ModmanageScreen:init()
+
+    resizable = true,
+    autoarrange_subviews=false,
+}
+
+local function get_mod_id_and_version(path)
+    local idfile = path .. '/info.txt'
+    local ok, lines = pcall(io.lines, idfile)
+    if not ok then return end
+    local id, version
+    for line in lines do
+        if not id then
+            _,_,id = line:find('^%[ID:([^%]]+)%]')
+        end
+        if not version then
+            -- note this doesn't include the closing brace since some people put
+            -- non-number characters in here, and DF only reads the leading digits
+            -- as the numeric version
+            _,_,version = line:find('^%[NUMERIC_VERSION:(%d+)')
+        end
+        -- note that we do *not* want to break out of this loop early since
+        -- lines has to hit EOF to close the file
+    end
+    return id, version
+end
+
+local function add_mod_paths(mod_paths, id, base_path, subdir)
+    local sep = base_path:endswith('/') and '' or '/'
+    local path = ('%s%s%s'):format(base_path, sep, subdir)
+    if dfhack.filesystem.isdir(path) then
+        table.insert(mod_paths, {id=id, path=path})
+    end
+end
+
+local function getWorldModlist()
+    -- ordered map of mod id -> {handled=bool, versions=map of version -> path}
+    local mods = utils.OrderedTable()
+    local mod_paths = {}
+
+    -- if a world is loaded, process active mods first, and lock to active version
+    if dfhack.isWorldLoaded() then
+        for _,path in ipairs(df.global.world.object_loader.object_load_order_src_dir) do
+            path = tostring(path.value)
+            -- skip vanilla "mods"
+            if not path:startswith(INSTALLED_MODS_PATH) then goto continue end
+            local id = get_mod_id_and_version(path)
+            if not id then goto continue end
+            mods[id] = {handled=true}
+            add_mod_paths(mod_paths, id, path, '.')
+            ::continue::
+        end
+        local modlist = {}
+        for _,mod in ipairs(mod_paths) do
+            table.insert(modlist,mod.id)
+        end
+        return modlist
+    end
+    qerror('No world is loaded')
+end
+
+function ModlistMenu:init()
+    local modlist = widgets.List{
+        view_id='modlist',
+        frame = {t=3},
+        choices = getWorldModlist()
+    }
+    self:addviews{
+        widgets.Label{
+            frame = { l=0, t=0 },
+            text = {'Active mods:'},
+        },
+        widgets.HotkeyLabel{
+            view_id='copy',
+            frame={t=1, r=1},
+            label='Copy modlist to clipboard',
+            text_pen=COLOR_YELLOW,
+            auto_width=true,
+            on_activate=function()
+                local mods = ''
+                for _,mod in ipairs(getWorldModlist()) do
+                    mods = (mods == '' and mod) or (mods .. ', ' .. mod)
+                end
+                dfhack.internal.setClipboardTextCp437(mods)
+            end,
+            enabled=function() return #modlist:getChoices() > 0 end,
```

Comment:
this is where the `view_id` actually comes in handy. You can declare the `widgets.List` inline instead of storing it in a local variable and change this line to be
```lua
enabled=function() return #self.subviews.modlist:getChoices() > 0 end,
```